spring-cloud-commons/src/main/java/org/springframework/cloud/commons/publisher/FluxFirstNonEmptyEmitting.java

```diff
@@ -325,6 +344,31 @@ public void onComplete() {
 			}
 		}
 
+		@Override
+		public String stepName() {
```

Comment:
this generic code has proven to be buggy. best approach is to override the method to return a constant. the goal of `stepName` is to give a human-friendly representation of the operator, which can be used eg. when printing additional backtraces in debug mode.

Reply:
so here we could have simply `return "firstNonEmpty";`, or perhaps `CloudFlux.firstNonEmpty` to make it obvious this is a cloud-custom operator?